code/game/objects/structures/headpike.dm

```diff
@@ -6,22 +6,29 @@
 	density = FALSE
 	anchored = TRUE
 	var/bonespear = FALSE
+	var/bamboospear = FALSE
 	var/obj/item/spear/spear
 	var/obj/item/bodypart/head/victim
 
 /obj/structure/headpike/bone //for bone spears
 	icon_state = "headpike-bone"
 	bonespear = TRUE
 
+/obj/structure/headpike/bamboo //for bamboo spears
+	icon_state = "headpike-bamboo"
+	bamboospear = TRUE
+
 /obj/structure/headpike/CheckParts(list/parts_list)
-	..()
 	victim = locate(/obj/item/bodypart/head) in parts_list
 	name = "[victim.name] on a spear"
-	update_icon()
 	if(bonespear)
 		spear = locate(/obj/item/spear/bonespear) in parts_list
+	else if(bamboospear)
+		spear = locate(/obj/item/spear/bamboospear) in parts_list
```

Comment:
Why isn't this in child proc instead?

Reply:
Honestly I tried a few things to make it work nicer and they all had other issues. I'm aware it's a bit awkward but I'm not sure how to handle it better...

Reply:
remove the useless new var and move this proc to child.
This might require refactoring other headspikes too, if you are having problems ping me.

Reply:
6ed6079 An attempt has been made. I have no idea if this'll be good enough but it certainly feels less hacky.